2001-03-26  Jakub Jelinek  <jakub@redhat.com>

	* cp-tree.h (flag_new_abi): Define to 0 unconditionally.
	* decl2.c (_flag_new_abi): Renamed from flag_new_abi.
	(lang_decode_option): Error that we don't support -fnew-abi.

--- gcc/cp/cp-tree.h.jj	Mon Mar 26 14:02:39 2001
+++ gcc/cp/cp-tree.h	Mon Mar 26 15:37:37 2001
@@ -210,7 +210,8 @@ Boston, MA 02111-1307, USA.  */
 
 /* Nonzero to enable experimental ABI changes.  */
 
-extern int flag_new_abi;
+extern int _flag_new_abi;
+#define flag_new_abi 0
 
 /* Nonzero to use __cxa_atexit, rather than atexit, to register
    destructors for local statics and global objects.  */
--- gcc/cp/decl2.c.jj	Fri Mar 23 20:16:00 2001
+++ gcc/cp/decl2.c	Mon Mar 26 15:38:58 2001
@@ -438,7 +438,7 @@ int flag_weak = 1;
 
 /* Nonzero to enable experimental ABI changes.  */
 
-int flag_new_abi;
+int _flag_new_abi;
 
 /* Nonzero to use __cxa_atexit, rather than atexit, to register
    destructors for local statics and global objects.  */
@@ -657,13 +657,14 @@ lang_decode_option (argc, argv)
         }
       else if (!strcmp (p, "new-abi"))
 	{
-	  flag_new_abi = 1;
+	  error ("-fnew-abi not supported in gcc-2.96-RH");
+	  _flag_new_abi = 1;
 	  flag_do_squangling = 1;
 	  flag_vtable_thunks = 1;
 	}
       else if (!strcmp (p, "no-new-abi"))
 	{
-	  flag_new_abi = 0;
+	  _flag_new_abi = 0;
 	  flag_do_squangling = 0;
 	}
       else if ((option_value
